pkg/net/bitswap.go

```diff
@@ -187,58 +193,90 @@ func (c BitswapClient) SpadeTraversal(parent context.Context,
 	startTime := time.Now()
 
 	i := uint(0)
+	var blk blocks.Block
 	for {
-		// Retrieval
-		logger.Infof("retrieving %s\n", cidToRetrieve.String())
-		blk, err := c.RetrieveBlock(parent, target, network, bswap, cidToRetrieve)
-
-		if err != nil {
-			return task.NewErrorRetrievalResultWithErrorResolution(task.RetrievalFailure, err), nil
-		}
-
-		// Verify returned content hashes to the CID we're expecting
-		if !blk.Cid().Equals(cidToRetrieve) {
-			return task.NewErrorRetrievalResult(task.CIDMismatch,
-				errors.Errorf("retrieved cid does not match requested: %s, %s",
-					blk.Cid().String(), cidToRetrieve)), nil
+		for j, cidToRetrieve := range cidsToRetrieve {
+			// Retrieval
+			logger.Infof("retrieving %s\n", cidToRetrieve.String())
+			blk, err := c.RetrieveBlock(parent, target, network, bswap, cidToRetrieve)
+
+			if err != nil {
+				return task.NewErrorRetrievalResultWithErrorResolution(task.RetrievalFailure, err), nil
+			}
+
+			// Verify returned content hashes to the CID we're expecting
+			if !blk.Cid().Equals(cidToRetrieve) {
+				return task.NewErrorRetrievalResult(task.CIDMismatch,
+					errors.Errorf("retrieved cid does not match requested: %s, %s",
+						blk.Cid().String(), cidToRetrieve)), nil
+			}
+
+			// Wait until we are at max depth and tried all the challenges
+			if i == maxTraverseDepth && j == len(cidsToRetrieve)-1 {
+				var size = int64(len(blk.RawData()))
```

Comment:
@jcace  what was the point of this size? It seems like we only care about the last block looked at?

Reply:
just for additional context - it goes in the retrieval results. I think its fine just like this.